crates/primitives/src/bytes/serde.rs

```diff
@@ -53,13 +53,18 @@ impl<'de> serde::Deserialize<'de> for Bytes {
             }
         }
 
-        deserializer.deserialize_any(BytesVisitor)
+        if deserializer.is_human_readable() {
+            deserializer.deserialize_any(BytesVisitor)
+        } else {
+            deserializer.deserialize_byte_buf(BytesVisitor)
+        }
     }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
+    use bincode as _;
```

Comment:
Nit: I don't think this does anything

```suggestion
```

Reply:
Without those I got:
```
warning: external crate `bincode` unused in `alloy_primitives`: remove the dependency or add `use bincode as _;`
```
When building the tests without the `serde` feature.
I don't know how to prevent this warning. Is there a better way?

Reply:
Ah I see, it's fine then. The problem is that it gets included in the test binaries but not in the library, so it flags it as unused in the library.

Reply:
So we decided to have the `use` in? In both places?
What I don't really understand is why the same is not needed for `serde_json`. Seems like it's only used in the serde tests as well...

Reply:
It is: https://github.com/alloy-rs/core/blob/4f7a72c476de6b008be275daaa577ef9e89156e5/crates/primitives/src/lib.rs#L20-L24